luigi/contrib/hdf5.py

```diff
@@ -0,0 +1,592 @@
+    for multithreading. As pytables does not allow simultaneous access to a storage each operation is protected by
+    locking the resource and unlocking it after finishing using file locks.
+    Other targets requesting access will wait until it is freed.
+
+    :note:
+            When table is appended there is no atomicity of write operation. If you want atomic write operation consider
+            using a fixed storage or set append to False.
+    """
+
+    def __init__(self, store, key, append=True, index_cols=(), expected_rows=None, format="table", split=None,
+                 complib="blosc", selector="index_table"):
+        """
+
+        :param store: path to hdf5file will be created if not existent.
+        :param key: key or node inside the storage to write to.
+        :param index_cols: this columns will be fully indexed for faster searches.
+                           Consider indexing often queried columns.
+        :param append:  if True the created table will be appendable. Only usable if table is openened in append mode.
+                        Multiple class to write will then append to the table. Default is True.
+        :param expected_rows: for faster I/O a good estimate of the table size should be estimated.
+        :param format:  either 'fixed' or 'table'
+        :param complib: which compression library to use. Complevel is default set to 1 as results
+                        do not differ to much using higher compression
+        :param split:  A dictionary or an integer useful to split tables along their columns. If a dictionary is passed
+                       it should describe a mapping between tables and columns. A table with the passed selector should
+                       be present in the keys. All indexed columns should be mapped to the selector table. Passing an
+                       integer will automatically create a valid split dictionary, where the number specifies the number
+                       of desired tables.
+        :param format: Only useful if split is not None. Specifies the selector table. This key must be present in the
+                       split dictionary.
+        :return: A target object
+        """
+        super(Hdf5TableTarget, self).__init__(key)
+        self.path = key
+        self.filesystem = Hdf5FileSystem(store)
+        self.table_params = dict(
+            store=store,
+            key=key,
+            append=append,
+            index_cols=index_cols,
+            expected_rows=expected_rows,
+            format=format,
+            split=split,
+            complib=complib,
+            selector=selector
+        )
+        # check if parameters are ok
+        AtomicHdf5Table.check_integrity(**self.table_params)
+
+    @property
+    def fs(self):
+        return self.filesystem
+
+    @property
+    def fn(self):
+        return self.table_params["store"]
```

Comment:
This one might be worth keeping, but I would rename it to `filename` and document it.